subprojects/core/src/main/java/org/gradle/api/internal/plugins/DefaultPluginManager.java

```diff
@@ -255,17 +256,44 @@ public void run(BuildOperationContext context) {
             return computeApplyPluginBuildOperationDetails(plugin);
         }
 
-        private BuildOperationDescriptor.Builder computeApplyPluginBuildOperationDetails(PluginImplementation<?> pluginImplementation) {
-            String identifier;
+        private BuildOperationDescriptor.Builder computeApplyPluginBuildOperationDetails(final PluginImplementation<?> pluginImplementation) {
+            String pluginIdentifier;
             if (pluginImplementation.getPluginId() != null) {
-                identifier = pluginImplementation.getPluginId().toString();
+                pluginIdentifier = pluginImplementation.getPluginId().toString();
             } else {
-                identifier = pluginImplementation.asClass().getName();
+                pluginIdentifier = pluginImplementation.asClass().getName();
             }
-            String name = "Apply plugin " + identifier;
+            String name = "Apply plugin " + pluginIdentifier;
+            final ConfigurationTargetIdentifier targetIdentifier = target.getConfigurationTargetIdentifier();
             return BuildOperationDescriptor.displayName(name + " to " + target.toString())
                 .name(name)
-                .details(new ApplyPluginBuildOperationType.DetailsImpl(pluginImplementation.getPluginId(), pluginImplementation.asClass().getName()));
+                .details(new ApplyPluginBuildOperationType.Details() {
```

Comment:
Readability mostly. And consistency, all the other details implementation live with their respective type.